libcaf/libcaf/repository.py

```diff
@@ -560,7 +533,7 @@ def diff(self, target1: Ref | Path | None = None, target2: Ref | Path | None = N
 
                         if record1.type == TreeRecordType.TREE:
                             try:
-                                stack.append((self._load_tree(record1.hash, tree_hashes), None, local_diff))
+                                stack.append((self._load_tree(record1.hash, tree_hashes1), None, local_diff))
```

Comment:
You didn't like my idea of hiding the tree cache in a class?

Reply:
Sure I liked the idea but it's a matter of preference, it's not a real issue with the code